src/input_control.js

```diff
@@ -18,15 +18,29 @@ module.exports = function (container, directions) {
             if (d3.event.keyCode === 13) {
                 d3.event.preventDefault();
 
-                directions
-                    .setOrigin(originInput.property('value'))
-                    .setDestination(destinationInput.property('value'));
+                geocode(function(err) {
+                    if (err) return this.fire('error', err);
 
-                if (directions.queryable())
-                    directions.query();
+                    if (directions.queryable())
+                        directions.query();
+                });
             }
         });
 
+    var geocode = control.geocode = function(cb) {
+        directions.geocode(originInput.property('value'), L.bind(function(err, res) {
```

Comment:
This file uses the module pattern, so `L.bind` is not necessary here (and in fact you don't pass a context parameter anyway). Instead of using `this`, use `control`. But note that `control` doesn't have a `fire` method -- we need to find some other way of reporting errors.